### PR TITLE
Fix hub service client query paging, and change device clean-up

### PR DESCRIPTION
**Pull request by drwill-ms**
Devices clean-up broken due to 2 bugs, one in query paging, the other in the clean-up app.

Also simplified clean-up to work on an exclusion list for deletion instead of an inclusion list for deletion.

### CODE REVIEW

File: e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1 [hunk 1]
```diff
@@ -221,7 +221,7 @@ $verificationCertPath = "$PSScriptRoot/verification.cer";
 
 $iotHubX509DeviceCertCommonName = "iothubx509device1";
 $iotHubX509DevicePfxPath = "$PSScriptRoot/IotHubX509Device.pfx";
-$iotHubX509CertChainDeviceCommonName = "iothubx509chaindevice1";
+$iotHubX509CertChainDeviceCommonName = "Save_iothubx509chaindevice1";
```
Comment (drwill-ms) [inline e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1:224]: Prefixing all static resources with "Save_"

File: iothub/service/samples/how to guides/CleanupDevicesSample/CleanupDevicesSample.cs [hunk 2]
```diff
@@ -93,31 +93,31 @@ private async Task CleanupDevices(int deviceCount)
                     } while (jobTimer.Elapsed < s_maxJobDuration);
 
                     // Wait until job is finished.
-                    IotHubJobResponse jobToBeDeleted = null;
                     jobTimer.Restart();
+                    IotHubJobResponse deleteDevicesJob = null;
                     while (jobTimer.Elapsed < s_maxJobDuration)
                     {
-                        jobToBeDeleted = await _hubClient.Devices.GetJobAsync(importDevicesToBeDeletedJob.JobId);
```
Comment (drwill-ms) [inline iothub/service/samples/how to guides/CleanupDevicesSample/CleanupDevicesSample.cs:100]: This was the bug. The value we got back from the method wasn't the one we checked for status on 101 and later.

File: iothub/service/src/Query/QueryClient.cs [hunk 3]
```diff
@@ -120,7 +120,7 @@ public virtual async Task<QueryResponse<T>> CreateAsync<T>(string query, QueryOp
                     request.Headers.Add(PageSizeHeader, options.PageSize.ToString());
                 }
 
-                HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                using HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
```
Comment (drwill-ms) [inline iothub/service/src/Query/QueryClient.cs:123]: Leak was here.

File: iothub/service/src/Query/QueryResponse.cs [hunk 4]
```diff
@@ -173,15 +175,23 @@ public async Task<bool> MoveNextAsync(QueryOptions queryOptions = default, Cance
...
             if (!string.IsNullOrEmpty(_originalQuery))
             {
-                QueryResponse<T> response = await _client.CreateAsync<T>(_originalQuery, queryOptionsClone, cancellationToken);
+                QueryResponse<T> response = await _client
+                    .CreateAsync<T>(_originalQuery, queryOptionsClone, cancellationToken)
+                    .ConfigureAwait(false);
                 CurrentPage = response.CurrentPage;
-                Current = CurrentPage.GetEnumerator().Current;
+                _items.Reset();
+                _items.MoveNext();
```
Comment (drwill-ms) [inline iothub/service/src/Query/QueryResponse.cs:183]: The next page was null until I added these calls.

**Review by timtay-microsoft — APPROVED** [review]

File: iothub/service/src/Query/QueryResponse.cs (hunk 4 above)
Comment (drwill-ms) [reply iothub/service/src/Query/QueryResponse.cs:183]: I should say the next page's `Current` item was null.

**Review by abhipsaMisra — APPROVED** [review]

**Review by timtay-microsoft — CHANGES_REQUESTED** [review]
I'm adding a test for iterating by item over multiple pages and I'm seeing a single result returned multiple times. Let me rework the query fix for you a bit before you merge this

File: iothub/service/samples/how to guides/CleanupDevicesSample/CleanupDevicesSample.cs [hunk 5]
```diff
@@ -193,7 +210,7 @@ private static Uri GetStorageAccountSasUriForCleanupJob(BlobContainerClient blob
                 | BlobContainerSasPermissions.Read
                 | BlobContainerSasPermissions.Delete;
 
-            var sasBuilder = new BlobSasBuilder(sasPermissions, DateTimeOffset.UtcNow.AddHours(1))
+            var sasBuilder = new BlobSasBuilder(sasPermissions, DateTimeOffset.UtcNow.AddHours(12))
```
Comment (drwill-ms) [inline iothub/service/samples/how to guides/CleanupDevicesSample/CleanupDevicesSample.cs:213]: We monitor the job for up to 4 hours, but it'll die sooner if it loses access to storage.

**Review by timtay-microsoft — APPROVED** [review]

**Review by patilsnr — APPROVED** [review]

**Pull request merged**